lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -166,3 +170,119 @@ def post(self, request, *args, **kwargs):
         # Populate user activity for tracking progress towards a user's course goals
         UserActivity.record_user_activity(user, course_key)
         return Response(status=(200))
+
+
+@view_auth_classes(is_authenticated=False)
+class BlocksInfoInCourseView(BlocksInCourseView):
+    """
+    **Use Case**
+
+        Returns the blocks in the course according to the requesting user's access level.
+        Add to response info fields with information about course
+
+    **Example requests**:
+
+        This api works with all versions {api_version}, you can use: v0.5, v1, v2 or v3
+
+        GET /api/mobile/{api_version}/course_info/blocks/?course_id=<course_id>
+        GET /api/mobile/{api_version}/course_info/blocks/?course_id=<course_id>
+            &username=anjali
+            &depth=all
+            &requested_fields=graded,format,student_view_multi_device,lti_url
+            &block_counts=video
+            &student_view_data=video
+            &block_types_filter=problem,html
+
```

Comment:
Can you add information about the params this view accpets?

Reply:
Yes, I have done that

Reply:
You can refer to the parent class about all params not used here. We are only using course_id and return_type here.